packages/zosuss/src/SshBaseHandler.ts

```diff
@@ -58,22 +59,108 @@ export abstract class SshBaseHandler implements ICommandHandler {
+                            await ConnectionPropsForSessCfg.addPropsOrPrompt<ISshSession>(
+                                sshSessCfgWithCreds,
+                                commandParameters.arguments,
+                                {
+                                    parms: commandParameters,
+                                    propertyOverrides: sshSessCfgOverride,
+                                    supportedAuthTypes: [
+                                        SessConstants.AUTH_TYPE_BASIC,
+                                    ],
+                                    propsToPromptFor: [
+                                        {
+                                            name: "keyPassphrase",
+                                            isGivenValueValid: (givenValue: {
+                                                [key: string]: any;
+                                            }) => {
+                                                let saveKP: boolean = true;
+                                                const result = utils.parseKey(
+                                                    fs.readFileSync(
+                                                        sshSessCfgWithCreds[
+                                                            "privateKey"
+                                                        ]
+                                                    ),
+                                                    givenValue.keyPassphrase
+                                                );
+                                                if (result instanceof Error)
+                                                    saveKP =
+                                                        !result.message.includes(
+                                                            "no passphrase given"
+                                                        ) &&
+                                                        !result.message.includes(
+                                                            "bad passphrase"
+                                                        );
+                                                return saveKP;
+                                            },
+                                        },
+                                    ],
+                                }
+                            );
+                        this.mSession = new SshSession(sshSessCfgWithCreds);
+                        await this.processCmd(commandParameters);
+                        success = true;
+                    } catch (retryError) {
+                        this.console.log(
+                            "\n" +
+                                `Key passphrase authentication failed! (${
+                                    attempt + 1
+                                }/${maxAttempts})` +
+                                "\n"
+                        );
+                        attempt++;
```

Comment:
The `attempt` variable could be incremented directly inside the string interpolation expression:
```suggestion
                        this.console.log(
                            "\n" +
                                `Key passphrase authentication failed! (${
                                    ++attempt
                                }/${maxAttempts})` +
                                "\n"
                        );
```

Not necessarily suggesting to change this, may want to keep the original syntax if you prefer incrementing on a separate line for readability 😋